HaskellVimNow.hs

```diff
@@ -0,0 +1,833 @@
+#!/usr/bin/env stack
+{- stack
+  script
+  --resolver lts-8.14
+  --package aeson
+  --package ansi-terminal
+  --package directory
+  --package foldl
+  --package managed
+  --package mtl
+  --package process
+  --package stache
+  --package system-filepath
+  --package text
+  --package time
+  --package transformers
+  --package turtle
+-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+
+import Control.Applicative ((<|>), empty, optional)
+import Control.Exception (bracket_, throwIO)
+import qualified Control.Foldl as Foldl
+import Control.Monad (filterM, guard, mfilter, unless, void, when)
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad.Managed (runManaged)
+import Control.Monad.Reader (MonadReader, ask, asks, runReaderT)
+import Control.Monad.Trans.Maybe (MaybeT(..), runMaybeT)
+import Data.Aeson ((.=), object)
+import Data.Foldable (forM_)
+import Data.Functor (($>))
+import Data.Maybe (isJust, isNothing, listToMaybe)
+import Data.Monoid ((<>))
+import qualified Data.Text
+import Data.Text (Text)
+import qualified Data.Text.IO
+import Data.Text.Lazy (toStrict)
+import Data.Time.Format (defaultTimeLocale, formatTime)
+import Data.Time.LocalTime (utcToLocalZonedTime)
+import Filesystem.Path.CurrentOS (FilePath, (</>))
+import qualified Filesystem.Path.CurrentOS as FS
+import Prelude hiding (FilePath)
+import qualified System.Console.ANSI as ANSI
+import qualified System.Directory
+import qualified System.Exit
+import qualified System.IO
+import System.IO.Error
+       (catchIOError, isDoesNotExistError, isPermissionError)
+import System.Info (os)
+import qualified System.Process
+import Text.Mustache (Template, renderMustache)
+import Text.Mustache.Compile.TH (mustache)
+import Turtle ((.||.))
+import qualified Turtle
+
+data HvnConfig = HvnConfig
+  { hvnCfgBasic :: Bool
+  , hvnCfgRepo :: Text
+  , hvnCfgBranch :: Text
+  , hvnCfgHome :: FilePath
+  , hvnCfgDest :: FilePath
+  , hvnCfgHoogleDb :: Bool
+  , hvnCfgHelperBinaries :: Bool
+  } deriving (Show)
+
+data HvnArgs = HvnArgs
+  { hvnArgsBasic :: Bool
+  , hvnArgsRepo :: Maybe Text
+  , hvnArgsBranch :: Maybe Text
+  , hvnArgsNoHoogleDb :: Bool
+  , hvnArgsNoHelperBinaries :: Bool
+  } deriving (Show)
+
+data HvnPackageManager
+  = HvnPackageManager'Brew
+  | HvnPackageManager'Dnf
+  | HvnPackageManager'Yum
+  | HvnPackageManager'Apt
+  | HvnPackageManager'Port
+  | HvnPackageManager'Chocolatey
+  | HvnPackageManager'Other
+  deriving (Show)
+
+defaultRepo :: Text
+defaultRepo = "https://github.com/jship/haskell-vim-now.git"
```

Comment:
I tried rebuilding on CI without cache, but the build still fails. I think the problem is this default repo path. It has already cloned the repo from my account, and then the new path conflicts:

```
The source repository path [https://github.com/jship/haskell-vim-now.git] does not match the
origin repository of the existing installation [https://github.com/begriffs/haskell-vim-now.git].
Please remove the existing installation [/home/travis/.config/haskell-vim-now] and try again.
```